pkg/k8s/watchers/pod.go

```diff
@@ -68,9 +78,86 @@ const podApiGroup = resources.K8sAPIGroupPodV1Core
 
 var ciliumEndpointSyncPodLabelsControllerGroup = controller.NewGroup("sync-pod-labels-with-cilium-endpoint")
 
+type k8sPodWatcherParams struct {
+	cell.In
+
+	K8sEventReporter *K8sEventReporter
+
+	Clientset         k8sClient.Clientset
```

Comment:
Depending on how often this set of arguments are repeated, do you feel its worth tidying them all up into a common sub-param structure?

Reply:
Good idea. But the only two dependencies that are actually used by all extracted k8s watchers are the following

```
K8sResourceSynced *k8sSynced.Resources
K8sAPIGroups      *k8sSynced.APIGroups
```

`Resources`, `ClientSet` & `K8sEventReporter` are only used by a subset (not even consistent). And there are also many other (already extracted and modularized) k8s watchers in the code that don't use such a common structure either. 

As much as i like this proposal; with the outlook that many of these watchers will be further refactored in the future anyway, i'd like to keep it this way and prevent further coupling between the watchers. And perhaps revisit this point in a later point in time.

Reply:
sounds good to me.